### PR TITLE
(SIMP-3951) Allow ignore to match interfaces

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Updated to match the 'ignore' parameter on input and output interfaces
* Disabled the scanblock tests since that is a different PR
* This change should be tested against the rest of the codebase prior
  to push

SIMP-3962 #close
SIMP-3963 #close

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-
-* table: The name of the table to which you are adding the rule. Default: 'filter'
-* first: If `true`, prepend the rule to the rule set. Default: `false`
-* absolute: If `true`, ensure that the rule is at the absolute beginning or end of the rule set. If multiple rules are marked as absolute then they are sorted by resource name. Default: `false`
-* order: The order in which the rule should appear. 1 is the minimum and 999 is the maximum. Default: '11'
-* resolve: If `true`, Puppet will resolve any hostnames in the rules prior to application to prevent iptables from starting should a name change in the future. Default: `true`
-* content: The full content of the rule that should be added to the rule set. You may place *anything* here. Required
-
-#### iptables_optimize
-#### ip6tables_optimize
```
Comment (jeefberkey) [inline README.md]: This is a good place for the 30,000 ft overview of the module. Technical documentation can still be looked for in the generated docs.

File: spec/fixtures/hieradata/hiera.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 :backends:
 - yaml
 :yaml:
-  :datadir: "/home/nick/simpdev/pupmod-simp-iptables/spec/fixtures/hieradata"
```
Comment (jeefberkey) [inline spec/fixtures/hieradata/hiera.yaml]: Whoops, this probably shouldn't be checked in

File: spec/acceptance/suites/default/00_default_spec.rb [hunk 3]
```diff
@@ -69,15 +69,15 @@ class { 'iptables': scanblock => true}
 EOM
       }
 
-      it 'should work with no errors' do
+      xit 'should work with no errors' do
```
Comment (jeefberkey) [inline spec/acceptance/suites/default/00_default_spec.rb]: What is `xit`?

**Comment by nick-markowski** [comment]
i'm writing some extra tests for this, hold off on merge

**Comment by trevor-vaughan** [comment]
@nick-markowski It's a Ruby regex, not a glob. Try `eth.*`.

**Comment by nick-markowski** [comment]
@trevor-vaughan still fails. In fact, if I make it ['eth.*','lo'], it still does not match properly and erases all of the manual rules

File: README.md [hunk 4]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
 
 ## Reference
 
-<<<<<<< be8c418e96cf122b1636397d7921baf0d5b33b86
-### Native Types
-
-#### `iptables_rule`
-
-A managed iptables rule. This may be used to provide ultimate flexibility in managing the system rules.
```
Comment (jeefberkey) [inline README.md]: Leave this in

File: README.md (hunk 1 above)
Comment (jeefberkey) [reply README.md]: Please leave this section in

File: README.md [hunk 5]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-#### iptables_optimize
-#### ip6tables_optimize
-
-This type collects all of the `iptables_rule` resources and compiles the final iptables rule set. By default, it will collapse rules into a single rule when possible to make the iptables rule set more efficient. Eventually, it will support `ipset`.
-
-The ip6tables_optimize type has the exact same parameters but affects ip6tables instead of iptables.
-
-##### Parameters
-
-* name: An arbitrary name for the resource. This resource is not meant to be called more than once...
```
Comment (jeefberkey) [inline README.md]: I don't see why this should go, but if you want to make the readme smaller I guess they *could* go read the type and see what params it needs.

File: README.md [hunk 6]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-* disable: If `true`, disable the management of iptables altogether. Default: `false`
-* ignore: Ignore all *running* iptables rules matching one or more provide Ruby regular expressions. The regular expressions are compared against both the JUMP and CHAIN options of the running rules. Anything matching these rules are excluded from the synchronization comparison against the new rules.
-
-  **Caveats**
-
-  Do NOT include the beginning and trailing slashes in your regular expressions.
```
Comment (jeefberkey) [inline README.md]: Leave this in

File: README.md [hunk 7]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-
-  If a rule has been added or removed, this setting ignored and iptables *will*
-be restarted! If you have services which are affected by this, make sure that
-they subscribe to Service['iptables'] and/or Service['ip6tables'] as
-appropriate.
-
-  **Examples**
-  ```
-  # Preserve all rules whose jump or chain begins with the word 'foo'
```
Comment (jeefberkey) [inline README.md]: Leave this in

File: README.md [hunk 8]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-* ip_list_hash_size: The hash table size. 0 means to calculate it based on ip_list_tot. Default: '0'
-* ip_list_perms: Permissions for /proc/net/xt_recent/* files. Default: '0640'
-* ip_list_uid: Numerical UID for ownership of /proc/net/xt_recent/* files. Defafult: '0'
-* ip_list_gid: Numerical GID for ownership of /proc/net/xt_recent/* files. Defafult: '0'
-
-### Classes
-
-#### `iptables`
-
-This sets the system up in a way that will maximally utilize the iptables native types.
```
Comment (jeefberkey) [inline README.md]: This needs explaining

File: README.md [hunk 9]
```diff
@@ -146,197 +148,6 @@ iptables::rule { 'example':
...
-
-##### Parameters
-
-* `authoritative`: If true, only iptables rules set by Puppet may be present on the system. Otherwise, only manage the *chains* that Puppet is managing.  Default: true
```
Comment (jeefberkey) [inline README.md]: My comments from above should be reflected for each respective section below

Comment (jeefberkey) [reply README.md]: Example are good, overview text is good, parameters are ok.

**Comment by jeefberkey** [comment]
This PR can be continued in https://simp-project.atlassian.net/browse/SIMP-4125

**Comment by trevor-vaughan** [comment]
@nick-markowski Investigating. Thanks for adding the tests, it makes things a LOT easier.

**Comment by trevor-vaughan** [comment]
@nick-markowski Turns out that `iptables-save` sorts based on the target chain and so we're currently comparing two different (but valid) output files.

File: CHANGELOG [hunk 10]
```diff
@@ -1,3 +1,11 @@
+* Thu Nov 30 2017 Trevor Vaughan <tvaughan@onyxpoint.com> - 6.1.1-0
```
Comment (nick-markowski) [inline CHANGELOG]: 6.1.0 has not been released yet, why is this bumped more than a Minor version?

**Pull request merged**